src/Health/Tests/Connection.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Tribe\Extensions\Promoter\Health\Tests;
+
+use Firebase\JWT\JWT;
+use Tribe\Extensions\Promoter\Health\Critical_Exception;
+use Tribe__Promoter__Connector;
+
+class Connection implements Runnable {
+	/**
+	 * @var Tribe__Promoter__Connector
+	 */
+	private $connector;
+	/**
+	 * @var array|\WP_Error
+	 */
+	private $response;
+	/**
+	 * @var string
+	 */
+	private $body;
+	/**
+	 * @var int
+	 */
+	private $code;
+	/**
+	 * @var string
+	 */
+	private $token = '';
+
+	public function run() {
+		$this->connector = tribe( 'promoter.connector' );
+
+		if ( ! $this->connector instanceof Tribe__Promoter__Connector ) {
+			throw new Critical_Exception( __( 'The connector class is not defined.', 'tribe_extensions_promoter' ) );
+		}
+
+		$license_info = tribe( 'promoter.pue' )->get_license_info();
+
+		if ( empty( $license_info['key'] ) ) {
+			throw new Critical_Exception( __( 'The license key of promoter is not present.', 'tribe_extensions_promoter' ) );
+		}
+
+		$payload = [
+			'licenseKey' => $license_info['key'],
+		];
+
+		$this->token = JWT::encode( $payload, get_option( 'tribe_promoter_auth_key' ) );
```

Comment:
`tribe_get_option()`?

Reply:
I think `tribe_get_option` won't work on this case as the tribe option are saved in a single field in this particular case this is stored in a separate field from tribe options.